Code/src/main/java/nl/utwente/group10/haskell/type/FuncT.java

```diff
@@ -30,4 +30,10 @@ public final String toHaskellType() {
         out.append(")");
         return out.toString();
     }
+
+    @Override
+    public FuncT getFresh() {
+        Type[] args = getFreshArgs();
+        return new FuncT(args[0], args[1]);
+    }
```

Comment:
I'm having second thoughts on the FooT classes. I originally thought of them as just convenient places to put some shorthand constructors (`new FuncT(intT, intT)` reads better than `new ConstT("->", intT, intT)`, but the identity of these objects has since been used to decide on the string representation, and I believe frontend does an `instanceof` check somewhere that doesn't really gel with that.

Would it make sense to *not* overwrite getFresh() for every class, throw around plain old ConstT's instead, and make the toString on ConstT itself smart enough to handle the special cases?

(The same goes for ListT and TupleT of course.)